### ISSUE TITLE
False positive for `try-except-raise` when a superclass exception is raised

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#5889]
### Bug description

```python
class MyException(ArithmeticError, ValueError):
    pass


try:
    raise MyException
except ArithmeticError:  # emits try-except-raise, but removing this handler is a behavior change
    raise
except ValueError:
    pass
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:7:0: W0706: The except handler raises immediately (try-except-raise)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 8.75/10, +0.00)
```

### Expected behavior

No message. Unfortunately I'm not sure how to solve what looks like a fundamental design problem.

### Pylint version

```shell
pylint 2.13.0-dev0
astroid 2.10.1-dev0
Python 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
```

**Comment by Pierre-Sassoulas** [comment]
My bad it still reproduce.
